### ISSUE TITLE
Wrong usage of String.replaceAll in NumberExtractor

### ISSUE THREAD

**Issue by rmanitz** [plutext/docx4j#340]
The code of the following method fails for GERMAN setup, where group seperator is "." (dot)

	/**
	 * Convert radix point to ".",
	 * and remove thousands separator,
	 * in preparation for our regex.
	 */
	private String prepare(String string) {
		string = string.replaceAll(groupingSeparator, "");
		
		if (decimalSymbol.equals(".")) {
			// do nothing
			return string;
		} else {
			return string.replace(decimalSymbol, ".");
		}
	}

Unfortunately, the following line replaces ALL characters of "string", because replaceAll uses the fist argument as a regexp (here  in the GERMAN case ".") that matches all characters of the string.

```
string = string.replaceAll(groupingSeparator, "");
```

In the british / USA case it works perfectly, because a regexp "," (comma) only matches the occurences of "," in the "string".

I'd like to suggest to replace this coding by something like that

```
if (".".equals(groupingSeperator)) {
   string = string.replaceAll("\\" + groupingSeparator, "");
} else {
   string = string.replaceAll(groupingSeparator, "");
}
```
(... or maybe a more efficient version of that ...)

Thank You in advance.

**Comment by plutext** [comment]
Thanks for this.  Implemented by https://github.com/plutext/docx4j/commit/fbec3169f9bd40287bd41ebe59c8d35c37347b62

**Comment by plutext** [comment]
Fix is in docx4j 6.1.2, which has now been released.
